feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/start/setupStakingType/SetupStakingTypeViewModel.kt

```diff
@@ -126,6 +133,26 @@ class SetupStakingTypeViewModel(
         }
     }
 
+    fun stakingTargetClicked(position: Int) {
+        launch {
+            val stakingType = stakingTypesDataFlow.first()[position]
+                .stakingTypeDetails
+                .stakingType
+
+            val selectionStore = editableSelectionStoreProvider.getSelectionStore(viewModelScope)
+
+            when {
+                stakingType.isDirectStaking() -> {
+                    setupStakingSharedState.set(SetupStakingProcess.ReadyToSubmit(selectionStore.getValidatorsOrEmpty(), SelectionMethod.CUSTOM))
+                    router.openSelectCustomValidators()
+                }
+                stakingType.isPoolStaking() -> {
+                    // TODO
+                }
+            }
+        }
```

Comment:
This should be hidden behind specific implementation of Provider interface. We do not want to expose staking types to viewModel untill its really necessart

Reply:
I will do it in the pr with pool logic to avoid conflicts: https://github.com/novasamatech/nova-wallet-android/pull/1091